src/objects/zcl_abapgit_object_aifc.clas.abap

```diff
@@ -0,0 +1,673 @@
+      IMPORTING
+        !ir_log           TYPE REF TO zif_abapgit_log
+      RETURNING
+        VALUE(rv_success) TYPE abap_bool
+      RAISING
+        zcx_abapgit_exception .
+    METHODS get_content_compress
+      IMPORTING
+        !ir_log     TYPE REF TO zif_abapgit_log
+        !is_ifkeys  TYPE ty_aif_key_s
+        !iv_package TYPE devclass
+      RAISING
+        zcx_abapgit_exception .
+    METHODS validate_interface
+      IMPORTING
+        !is_ifkeys        TYPE ty_aif_key_s
+      RETURNING
+        VALUE(rv_success) TYPE abap_bool
+      RAISING
+        zcx_abapgit_exception .
+    METHODS compress_interface
+      IMPORTING
+        !is_ifkeys        TYPE ty_aif_key_s
+        !ir_log           TYPE REF TO zif_abapgit_log
+      RETURNING
+        VALUE(rv_success) TYPE abap_bool
+      RAISING
+        zcx_abapgit_exception .
+  PRIVATE SECTION.
+
+    DATA mr_abapgit_util TYPE REF TO object.
+ENDCLASS.
+
+
+
+CLASS ZCL_ABAPGIT_OBJECT_AIFC IMPLEMENTATION.
+
+
+  METHOD authorization_check.
+    DATA: lv_meth       TYPE string,
+          ls_parameter  TYPE abap_parmbind,
+          lt_parameters TYPE abap_parmbind_tab.
+
+    DATA: lr_dyn_call_error TYPE REF TO cx_sy_dyn_call_error.
+    DATA: lr_root TYPE REF TO cx_root.
+
+    rv_success = abap_false.
+
+    TRY.
+        lv_meth     = '/AIF/IF_ABAPGIT_AIFC_UTIL~AUTHORIZATION_CHECK'.
+
+        ls_parameter-name = 'RV_SUCCESS'.
+        ls_parameter-kind = cl_abap_objectdescr=>returning.
+        GET REFERENCE OF rv_success INTO ls_parameter-value.
+        INSERT ls_parameter INTO TABLE lt_parameters.
+
+        CALL METHOD mr_abapgit_util->(lv_meth)
+          PARAMETER-TABLE
+          lt_parameters.
+
+      CATCH cx_sy_dyn_call_error INTO lr_dyn_call_error.
+        zcx_abapgit_exception=>raise( iv_text = TEXT-003
+                                      ix_previous = lr_dyn_call_error ).
+      CATCH cx_root INTO lr_root.
+        ir_log->add_exception( ix_exc  = lr_root
+                               is_item = ms_item ).
+        RETURN.
+    ENDTRY.
+
+  ENDMETHOD.
+
+
+  METHOD compress_interface.
+    DATA: lv_meth       TYPE string,
+          ls_parameter  TYPE abap_parmbind,
+          lt_parameters TYPE abap_parmbind_tab.
+
+    DATA: lr_dyn_call_error TYPE REF TO cx_sy_dyn_call_error.
+    DATA: lr_root TYPE REF TO cx_root.
+
+    TRY.
+        lv_meth     = '/AIF/IF_ABAPGIT_AIFC_UTIL~COMPRESS_INTERFACE'.
+
+        ls_parameter-name = 'IS_IFKEYS'.
+        ls_parameter-kind = cl_abap_objectdescr=>exporting.
+        GET REFERENCE OF is_ifkeys INTO ls_parameter-value.
+        INSERT ls_parameter INTO TABLE lt_parameters.
+
+        ls_parameter-name = 'RV_SUCCESS'.
+        ls_parameter-kind = cl_abap_objectdescr=>returning.
+        GET REFERENCE OF rv_success INTO ls_parameter-value.
+        INSERT ls_parameter INTO TABLE lt_parameters.
+
+        CALL METHOD mr_abapgit_util->(lv_meth)
+          PARAMETER-TABLE
+          lt_parameters.
```

Comment:
There's no need to use `PARAMETER-TABLE` in dynamic method call. Supplying parameters explicitly will make maintenance a bit easier and the code cleaner. What do you think?

This is e.g. how it's usually done in other places.
https://github.com/abapGit/abapGit/blob/fe363f7b0a9681302cc3ec320211c563bdb4f7b4/src/objects/zcl_abapgit_object_ddls.clas.abap#L141-L145

Reply:
Coding was adjusted as requested.